### PR TITLE
[Wrangler] Add information about wrangler login to a remote machine

### LINKED ISSUE #884: oauth redirect flow broken inside vm
I typically develop in a vagrant VM where there is no browser installed, the oauth flow in wrangler2 breaks because the redirect URI has "localhost:8976" set in the callback parameter.  We should be able to support copy pasting the callback URL to another machine and complete the auth flow there (cloudflared supports this)

Repro:
```
vagrant@kkoenig-devbox:~$ echo "export default { fetch() { return new Response('hello world') } }" > index.js
vagrant@kkoenig-devbox:~$  npx wrangler@beta dev index.js
 ⛅️ wrangler 0.0.27
--------------------
Attempting to login via OAuth...
Failed to open https://dash.cloudflare.com/oauth2/auth?response_type=code&client_id=... redirect_uri=http%3A%2F%2Flocalhost%3A8976%2Foauth%2Fcallback&scope=account%3Aread%20user%3Aread%20workers%3Awrite%20... in a browser
```

When I hope the URL in my host OS browser and complete the auth flow, the callback is never completed because the url is `http://localhost:8976/oauth/callback?...`

The same thing repros with wrangler 1
```
vagrant@kkoenig-devbox:~$ npx @cloudflare/wrangler login
Allow Wrangler to open a page in your browser? [y/n]
y
💁  Opened a link in your default browser: https://dash.cloudflare.com/oauth2/auth?redirect_url=<localhost>
```

But not with cloudflared which performs the same handshake
```
vagrant@kkoenig-devbox:~$ cloudflared login
Please open the following URL and log in with your Cloudflare account:

https://dash.cloudflare.com/argotunnel?callback=https%3A%2F%2Flogin.cloudflareaccess.org...

Leave cloudflared running to download the cert automatically.
You have successfully logged in.
```

### PR DESCRIPTION
**Pull request by cameron-robey**
Having had people ask for support multiple times (https://github.com/cloudflare/wrangler2/issues/884, https://github.com/cloudflare/wrangler2/issues/1885) with using wrangler on a remote server, it is best we add some documentation about a work around.

Closes https://github.com/cloudflare/wrangler2/issues/884

### CODE REVIEW

File: content/workers/wrangler/get-started.md [hunk 1]
```diff
@@ -73,6 +72,24 @@ If you prefer to use API tokens for authentication, such as in headless or conti
 
 Open the browser, log in to your account, and select **Allow**. This will send an OAuth Token to Wrangler so it can deploy your scripts to Cloudflare.
 
+### Using `wrangler login` in a remote machine
```
Comment (rozenmd) [inline content/workers/wrangler/get-started.md]:
```suggestion
### Using `wrangler login` on a remote machine
```

**Review by JacobMGEvans — APPROVED** [review]

**Review by rozenmd — CHANGES_REQUESTED** [review]
doing a review

File: content/workers/wrangler/get-started.md [hunk 2]
```diff
@@ -73,6 +72,24 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+If you are using wrangler from a remote machine but run the login flow from your local browser, you will get an error after logging in that `This site can't be reached`. To finish the login flow, you can use `curl` or an equivalent request library on the remote machine to fetch this `localhost` url.
```
Comment (rozenmd) [inline content/workers/wrangler/get-started.md]:
```suggestion
If you are using wrangler from a remote machine, but run the login flow from your local browser, after logging in you will see the following error message: `This site can't be reached`. 

To finish the login flow, you can use `curl` or an equivalent request library on the remote machine to fetch this `localhost` url.
```

File: content/workers/wrangler/get-started.md [hunk 3]
```diff
@@ -73,6 +72,24 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+```sh
+$ wrangler login
+ ⛅️ wrangler 2.1.6
+-------------------
+Attempting to login via OAuth...
+Opening a link in your default browser: https://dash.cloudflare.com/oauth2/auth?xyz...
+```
+
+In a second terminal session:
```
Comment (rozenmd) [inline content/workers/wrangler/get-started.md]:
```suggestion
In a second terminal session, run:
```

File: content/workers/wrangler/get-started.md [hunk 4]
```diff
@@ -73,6 +72,24 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+```
```
Comment (rozenmd) [inline content/workers/wrangler/get-started.md]:
```suggestion
```sh
```

File: content/workers/wrangler/get-started.md (hunk 1 above)
Comment (JacobMGEvans) [reply content/workers/wrangler/get-started.md]: Aren't these grammatically equivalent in this instance?

File: content/workers/wrangler/get-started.md (hunk 2 above)
Comment (JacobMGEvans) [reply content/workers/wrangler/get-started.md]: This makes sense and is more concise.

File: content/workers/wrangler/get-started.md (hunk 1 above)
Comment (rozenmd) [reply content/workers/wrangler/get-started.md]: You'll typically find `on a server/computer` to be the more common usage, `in a server/computer/remote machine` implies being physically inside it with the RAM/CPU

Comment (cameron-robey) [reply content/workers/wrangler/get-started.md]: https://github.com/cloudflare/cloudflare-docs/pull/5981/commits/76cacc2478f9d91726aa2579d6b5aef32fd9d6a8

File: content/workers/wrangler/get-started.md (hunk 2 above)
Comment (cameron-robey) [reply content/workers/wrangler/get-started.md]: https://github.com/cloudflare/cloudflare-docs/pull/5981/commits/76cacc2478f9d91726aa2579d6b5aef32fd9d6a8

File: content/workers/wrangler/get-started.md (hunk 4 above)
Comment (cameron-robey) [reply content/workers/wrangler/get-started.md]: https://github.com/cloudflare/cloudflare-docs/pull/5981/commits/76cacc2478f9d91726aa2579d6b5aef32fd9d6a8

File: content/workers/wrangler/get-started.md (hunk 3 above)
Comment (cameron-robey) [reply content/workers/wrangler/get-started.md]: https://github.com/cloudflare/cloudflare-docs/pull/5981/commits/76cacc2478f9d91726aa2579d6b5aef32fd9d6a8

File: content/workers/wrangler/get-started.md [hunk 5]
```diff
@@ -73,6 +72,24 @@ If you prefer to use API tokens for authentication, such as in headless or conti
 
 Open the browser, log in to your account, and select **Allow**. This will send an OAuth Token to Wrangler so it can deploy your scripts to Cloudflare.
 
+### Using `wrangler login` on a remote machine
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
### Use `wrangler login` on a remote machine
```

File: content/workers/wrangler/get-started.md [hunk 6]
```diff
@@ -73,6 +72,24 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+If you are using wrangler from a remote machine, but run the login flow from your local browser, after logging in you will see the following error message: `This site can't be reached`.
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
If you are using Wrangler from a remote machine, but run the login flow from your local browser, you will receive the following error message after logging in:`This site can't be reached`.
```

File: content/workers/wrangler/get-started.md [hunk 7]
```diff
@@ -73,6 +72,26 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+To finish the login flow, you can use `curl` or an equivalent request library on the remote machine to fetch this `localhost` url.
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
To finish the login flow, you can use `curl` or an equivalent request library on the remote machine to fetch this `localhost` URL.
```

**Review by deadlypants1973 — CHANGES_REQUESTED** [review]

File: content/workers/wrangler/get-started.md [hunk 8]
```diff
@@ -73,6 +72,26 @@ If you prefer to use API tokens for authentication, such as in headless or conti
 
 Open the browser, log in to your account, and select **Allow**. This will send an OAuth Token to Wrangler so it can deploy your scripts to Cloudflare.
 
+### Use `wrangler login` on a remote machine
+
+If you are using Wrangler from a remote machine, but run the login flow from your local browser, you will receive the following error message after logging in:`This site can't be reached`.
+
+To finish the login flow, you can use `curl` or an equivalent request library on the remote machine to fetch this `localhost` URL.
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
To finish the login flow, run `wrangler login` and go through the login flow in the browser:
```

File: content/workers/wrangler/get-started.md [hunk 9]
```diff
@@ -73,6 +72,26 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+```sh
+$ wrangler login
+ ⛅️ wrangler 2.1.6
+-------------------
+Attempting to login via OAuth...
+Opening a link in your default browser: https://dash.cloudflare.com/oauth2/auth?xyz...
+```
+
+In a second terminal session, run:
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
Leave the login flow active. Open a second terminal session. In that second terminal session, use `curl` or an equivalent request library on the remote machine to fetch this `localhost` URL:
```

**Review by deadlypants1973 — APPROVED** [review]
PCX review

File: content/workers/wrangler/get-started.md [hunk 10]
```diff
@@ -73,6 +72,26 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+```sh
+$ wrangler login
+ ⛅️ wrangler 2.1.6
+-------------------
+Attempting to login via OAuth...
+Opening a link in your default browser: https://dash.cloudflare.com/oauth2/auth?xyz...
+```
+
+Leave the login flow active. Open a second terminal session. In that second terminal session, use `curl` or an equivalent request library on the remote machine to fetch this `localhost` URL:
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
The browser login flow will redirect you to a `localhost` URL on your machine.

Leave the login flow active. Open a second terminal session. In that second terminal session, use `curl` or an equivalent request library on the remote machine to fetch this `localhost` URL. Copy and paste the `localhost` URL that was generated during the `wrangler login` flow and run:
```

File: content/workers/wrangler/get-started.md [hunk 11]
```diff
@@ -73,6 +72,26 @@ If you prefer to use API tokens for authentication, such as in headless or conti
...
+
+```sh
+$ curl http://localhost:8976/oauth/callback?code=xyz...
```
Comment (deadlypants1973) [inline content/workers/wrangler/get-started.md]:
```suggestion
$ curl <LOCALHOST_URL>
```

**Review by deadlypants1973 — APPROVED** [review]

**Review by rozenmd — APPROVED** [review]
🚢

**Pull request merged**